### ISSUE TITLE
 Error: Index passed  to line/column function (x) does not keep with order (last was x)

### ISSUE THREAD

**Issue by SerhiiBilyk** [htmllint/htmllint#248]
```html
<input type="text" readonly/>
```
Linter throws an error
```
Error: Index passed  to line/column function (322) does not keep with order (last was 453)
```
But if there  a space at the and of boolean attribute, 
 ```html
<input type="text" readonly />
```
It works Ok

I think the bug is inside attr_parse.js file.
 
```javascript
var attrRegex = /(\s*)([^ "'>=\^]+)((\s*=\s*)((?:"[^"]*")|(?:'[^']*')|(?:\S+)))?/g;
```
execute [ readonly/ ] attribute from
 ```html 
<input type="text" readonly/>
```
but must execute only [ readonly ]

it works as expected if I change
```javascript
 var attrRegex = /(\s*)([^ "'>=\^\/]+)((\s*=\s*)((?:"[^"]*")|(?:'[^']*')|(?:\S+)))?/g;
```

**Comment by ericcornelissen** [comment]
For you information, in the case of `<input>` in HTML5 the closing backslash should be omitted. So, yes the error is incorrect, but your solution is invalid (note here that this is _"An unofficial html5 linter and validator."_[[1]](https://github.com/htmllint/htmllint/blob/master/README.md)).

However, as per [this discussion](https://stackoverflow.com/questions/3558119/are-non-void-self-closing-tags-valid-in-html5) it is worth to note that in some cases the backslash may still be needed (e.g. in SVGs, relatedly #94). But I'm not familiar enough with the inner workings of this linter to tell if that is actually relevant...

**Comment by SerhiiBilyk** [comment]
I'm agree with you, but html-linter does not throw an error if there [closing backslash] inside input.
Linter throws "Index passed to line/column function (x) does not keep with order (last was x)". For me, this message is not clear.One more thing, when I use htmllinter with lint-staged package (https://github.com/okonet/lint-staged/issues/499) and htmllinter receives 'readonly/>' due to lint-staged there are no error at all. In this case, htmllinter behaves as html code is valid, even if there are non valid html elements, and all checks are succesfull.

**Comment by ericcornelissen** [comment]
> I'm agree with you, but html-linter does not throw special linter warning if there closing backslash inside input(< input readonly/>. Linter throws

I think it should, and indeed the message is not clear. However, I'm not a maintainer of this project so I can't decide on what will happen.

In hindsight, maybe calling your solution invalid wasn't completely fair. I think it will be useful to merge #249 so the attr-order check won't fail because of this 🤔

**Comment by SerhiiBilyk** [comment]
@ericcornelissen Who can submit  my PR?

**Comment by mlochbaum** [comment]
I think changing the regex is a reasonable fix for this. Per [the spec](https://html.spec.whatwg.org/multipage/syntax.html#attributes-2), slashes aren't allowed in attribute names, so this makes the attribute parsing more accurate.

This situation does only occur with invalid html (I could be wrong here, of course). However, htmllint tries to work with invalid html to the greatest extent possible. So it should indicate the issue (which it does, if `tag-self-close` is set to "never", i.e. html5 style) but otherwise handle it like a browser would.
